geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -4203,7 +4204,11 @@ public JSONFormatter getJsonFormatter() {
 
   @Override
   public QueryService getLocalQueryService() {
-    return new DefaultQueryService(this);
+    try {
+      return new DefaultQueryService(this);
+    } catch (ServiceNotFoundException e) {
+      throw new IllegalStateException(e.getMessage(), e.getCause());
+    }
```

Comment:
See my comments in the `DefaultQueryService` constructor regarding the new exceptions we are throwing.